lib/exceptions.h

```diff
@@ -31,8 +31,26 @@ constexpr char ANSI_BLUE[] = "\e[34m";
 constexpr char ANSI_CLR[]  = "\e[0m";
 
 /// Checks if stderr is redirected to a file
+/// Check is done only once and then saved to a static variable
 inline bool is_cerr_redirected() {
-    return ttyname(fileno(stderr)) == nullptr;  // NOLINT(runtime/threadsafe_fn)
+    static bool is_redir = ttyname(fileno(stderr)) == nullptr;  // NOLINT(runtime/threadsafe_fn)
```

Comment:
you are not saving anything, the call to ttyname is still made every time.
you need to also remember whether is_redir has been initialized, using a second static variable.